// @nrwl/react/src/schematics/storybook-configuration/configuration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.storybookConfigurationGenerator = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const ngcli_adapter_1 = require("@nrwl/devkit/ngcli-adapter");
function generateStories(schema) {
    return (tree, context) => {
        return schematics_1.schematic('stories', {
            project: schema.name,
            generateCypressSpecs: schema.configureCypress && schema.generateCypressSpecs,
            js: schema.js,
        });
    };
}
function default_1(schema) {
    return schematics_1.chain([
        schematics_1.externalSchematic('@nrwl/storybook', 'configuration', {
            name: schema.name,
            uiFramework: '@storybook/react',
            configureCypress: schema.configureCypress,
            generateCypressSpecs: schema.generateCypressSpecs,
            js: schema.js,
            linter: schema.linter,
        }),
        schema.generateStories ? generateStories(schema) : schematics_1.noop(),
    ]);
}
exports.default = default_1;
exports.storybookConfigurationGenerator = ngcli_adapter_1.wrapAngularDevkitSchematic('@nrwl/react', 'storybook-configuration');
